pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -355,13 +355,17 @@ func (gc *GarbageCollector) attemptToDeleteWorker() bool {
 		}
 		// retry if garbage collection of an object failed.
 		gc.attemptToDelete.AddRateLimited(item)
+		return true
 	} else if !n.isObserved() {
 		// requeue if item hasn't been observed via an informer event yet.
 		// otherwise a virtual node for an item added AND removed during watch reestablishment can get stuck in the graph and never removed.
 		// see https://issue.k8s.io/56121
 		klog.V(5).Infof("item %s hasn't been observed via informer yet", n.identity)
 		gc.attemptToDelete.AddRateLimited(item)
+		return true
 	}
+
+	gc.attemptToDelete.Forget(item)
```

Comment:
This change makes sense to me, but I'd like to see the if/else structure above turned into a switch/case with every case returning so that it's easier for me to read.

Reply:
looking at other return points that would be made more obvious in the switch/case, why don't we forget in all the cases where we return and have not added it back to the workqueue?

Reply:
Got it, updated. PTAL thanks.